mdata/aggmetric.go

```diff
@@ -66,6 +66,12 @@ func NewAggMetric(store Store, cachePusher cache.CachePusher, key string, retent
 	return &m
 }
 
+func (a *AggMetric) maybeSyncChunkSaveState(ts uint32) {
```

Comment:
i really dont like the name "maybeSyncChunkSaveState".  
I would much rather you just wrapped the calls to SyncChunkSaveState in  `if cntMetrc != nil `